src/transforms/transaction.rs

```diff
@@ -0,0 +1,813 @@
+use super::Transform;
+use crate::{
+    conditions::{AnyCondition, Condition},
+    event::discriminant::Discriminant,
+    event::{Event, LogEvent, Value},
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+};
+use async_stream::stream;
+use bytes::{Bytes, BytesMut};
+use chrono::{DateTime, Utc};
+use futures::{
+    compat::{Compat, Compat01As03},
+    stream,
+    stream::StreamExt,
+};
+use futures01::Stream as Stream01;
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+use std::collections::{hash_map, HashMap};
+use std::time::{Duration, Instant};
+use string_cache::DefaultAtom as Atom;
+
+//------------------------------------------------------------------------------
+
+#[derive(Deserialize, Serialize, Debug, Derivative)]
+#[serde(deny_unknown_fields, default)]
+#[derivative(Default)]
+pub struct TransactionConfig {
```

Comment:
I don't see any features of derivative being used in this struct. Does it handle defaults differently than rust's standard default?

Reply:
Looks unnecessary! Removed.